app/src/main/java/org/fossasia/openevent/app/qrscan/ScanQRActivity.java

```diff
@@ -251,4 +246,14 @@ public void startScan() {
     public void stopScan() {
         cameraSource.stop();
     }
+
+    public void showToggleDialog(long attendeeId) {
+        BottomSheetDialogFragment bottomSheetDialogFragment = AttendeeCheckInFragment.newInstance(attendeeId);
+        bottomSheetDialogFragment.show(getSupportFragmentManager(), bottomSheetDialogFragment.getTag());
+    }
+
+    @Override
+    public void onDialogDismissed() {
+        presenter.allowScan(true);
```

Comment:
I meant as soon as presenter triggers the dialog to be shown -> onScannedAttendee, you automatically set to false, then it won't create this problem